### PR TITLE
[doxygen] Update to 1.9.5. Fixes JB#59742

### PR DESCRIPTION
**Pull request by vigejolla**
Also changed to use submodule packaging and CMake.

Signed-off-by: Ville Nummela <ville.nummela@jolla.com>

### CODE REVIEW

**Comment by vigejolla** [comment]
I would have liked to have changes related to submodule packaging and cmake in separate commits, but that would have made things unnecessarily complex as the old version could not be built with submodule packaging, the old version could not be built with cmake and the new version could not be built without cmake.

File: rpm/doxygen.spec [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+%prep
+%autosetup -p1 -n %{name}-%{version}/doxygen
+
+%build
+unset QTDIR
+%cmake
+%make_build
+
+%install
+rm -rf %{buildroot}
```
Comment (martyone) [inline rpm/doxygen.spec]: It's deprecated to rm buildroot, %install already does it.

File: rpm/doxygen.spec [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+%make_install
+
+%files
+%defattr(-,root,root,-)
+%attr(755,root,root) /usr/bin/*
```
Comment (martyone) [inline rpm/doxygen.spec]: `%{_bindir}` should be used and it's common to omit `%attr` here - I believe the files are already installed with correct permissions by cmake.

**Review by martyone — APPROVED** [review]
Two suggestions + I wouldn't worry about build broken in mid of a PR. For me that's something to expect, so I would prefer to split the commits too.

File: rpm/doxygen.spec (hunk 1 above)
Comment (vigejolla) [reply rpm/doxygen.spec]: removed

File: rpm/doxygen.spec (hunk 2 above)
Comment (vigejolla) [reply rpm/doxygen.spec]: fixed. Also added the filename, there is only one binary.

**Pull request merged**